tests/unit/s2n_drbg_test.c

```diff
@@ -451,6 +451,8 @@ int main(int argc, char **argv)
     EXPECT_SUCCESS(s2n_stuffer_free(&nist_aes256_reference_entropy));
 
     EXPECT_SUCCESS(s2n_config_free(config));
-
+    /* Clean up with previously set functions */
+    POSIX_GUARD_RESULT(s2n_rand_cleanup());
```

Comment:
I think I understand what's happening here but want to make sure: it looks like `check_drgb_version` sets custom rand callbacks:
https://github.com/aws/s2n-tls/blob/ce0234e4116d05936c0847a1d3874f3851e44b3d/tests/unit/s2n_drbg_test.c#L298-L299

But since our default rand callback was already initialized, /dev/urandom was already opened. So when we go to cleanup in `END_TEST`, the `nist_fake_entropy_init_cleanup` callback is invoked, which does nothing. So /dev/urandom remains open. And this change sets the rand callbacks back to default so that the proper cleanup function is invoked. Is that correct?

If so, I think a better fix for this would be to properly call `s2n_rand_set_callbacks` before `s2n_init()`, like the documentation describes:
https://github.com/aws/s2n-tls/blob/ce0234e4116d05936c0847a1d3874f3851e44b3d/api/s2n.h#L586-L588

So the proposal would be to remove the `s2n_rand_set_callbacks()` call in `check_drgb_version()`, and instead call `s2n_cleanup()`, `s2n_rand_set_callbacks()`, and `s2n_init()` whenever a new rand callback is needed. Otherwise, this relies on `s2n_rand_cleanup()` being called twice (once manually and once in s2n_cleanup) in order to properly clean up its state, which seems wrong.